### ISSUE TITLE
PHP-Scoper 0.18.12 breaks PHAR for PHPUnit 8.5 on PHP 7.2

### ISSUE THREAD

**Issue by sebastianbergmann** [sebastianbergmann/phpunit#5855]
After updating PHP-Scoper from 0.18.11 to 0.18.12 in a3d40a3dbc1763c57000cb9406caca9a75ec91e7, the PHAR for PHPUnit 8.5 does not work with PHP 7.2:

```
PHP Parse error:  syntax error, unexpected end of file, expecting variable (T_VARIABLE) or heredoc end (T_END_HEREDOC) or ${ (T_DOLLAR_OPEN_CURLY_BRACES) or {$ (T_CURLY_OPEN) in phar:///home/runner/work/phpunit/phpunit/build/artifacts/phpunit-snapshot.phar/phpunit/Framework/MockObject/Generator.php on line 736
```

**Comment by sebastianbergmann** [comment]
@theofidry Before I dig into this, is there anything that would be obvious to you that might explain this? Thanks!

**Comment by sebastianbergmann** [comment]
@mfn PHP-Scoper requires a newer version of PHP, yes. But that does not have to mean that the modified code no longer works with PHP 7.2.

**Comment by theofidry** [comment]
Aside from the PHP-Parser upgrade I don't see anything that could cause this :/ Do you have the offending code by any chance? I can take a look at it

**Comment by sebastianbergmann** [comment]
```
$ git clone git@github.com:sebastianbergmann/phpunit.git
$ cd phpunit
$ git checkout 8.5
$ ant phar-snapshot
$ php-72 build/artifacts/phpunit-snapshot.phar

Parse error: syntax error in phar:///usr/local/src/phpunit/build/artifacts/phpunit-snapshot.phar/phpunit/Framework/MockObject/Generator.php on line 736
```
```
$ phar extract -f /usr/local/src/phpunit/build/artifacts/phpunit-snapshot.phar
$ php-72 -l phpunit/Framework/MockObject/Generator.php

Parse error: syntax error in phpunit/Framework/MockObject/Generator.php on line 736
Errors parsing phpunit/Framework/MockObject/Generator.php
```

Attached is the result of `ant phar-snapshot` for the current state of the `8.5` branch.

[phpunit-snapshot.phar.tar.gz](https://github.com/user-attachments/files/15741682/phpunit-snapshot.phar.tar.gz)

**Comment by theofidry** [comment]
Ok I got a small reproducer within the PHP-Scoper test suite, will look more into it

**Comment by theofidry** [comment]
@sebastianbergmann this should be fixed in 0.18.14

**Comment by sebastianbergmann** [comment]
Thank you!
